File: ecommerce/src/main/java/com/ecommerce/ecommerce/Service/ProductService.java
package com.ecommerce.ecommerce.Service;

import com.ecommerce.ecommerce.DTO.ProductDTO;
import com.ecommerce.ecommerce.DTO.ProductSaveDTO;
import com.ecommerce.ecommerce.DTO.ProductUpdateDTO;
import org.springframework.stereotype.Service;

import java.util.List;
@Service
public interface ProductService {
    String addCustomer(ProductSaveDTO productSaveDTO);

    List<ProductDTO> getAllProducts();

    String updateProduct(ProductUpdateDTO productUpdateDTO);

    boolean deleteCustomer(int product_id);


    List<ProductDTO> getProductsById(int product_id);
}

File: app/src/app/service/product.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Product } from '../model/product';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  baseUrl = "http://localhost:8080/api/v1/ecommerce"

  constructor(private httpClient: HttpClient) { }

  public addProduct1(productObject: any) {
    return this.httpClient.post<Product>(this.baseUrl + "/save", productObject);
  }
  
  public getProductById(product_id:any){
    return this.httpClient.get<Product>(this.baseUrl + "/getProductById/"+ product_id)
  }

  public getAllProduct(){
    return this.httpClient.get<Product[]>(this.baseUrl + "/getAllProducts")
  }

  public deleteProduct(product_id: any) {
    console.log("Detele",product_id);
    
    return this.httpClient.delete<Product>(this.baseUrl + "/delete/"+ product_id)
  }

  public updateProduct(productUpdateDTO: any): Observable<string> {
    console.log("Update Service")
    console.log("Update Service", this.baseUrl + "/update")
    console.log("Update Service", productUpdateDTO)
    return this.httpClient.put<string>(this.baseUrl + "/update", productUpdateDTO);
  }
}
